### PR TITLE
Move ntt to new dir. Add correctness tests.

### PR DESCRIPTION
**Pull request by cgyurgyik**

### CODE REVIEW

File: ntt/ntt-cooley-tukey.expect [hunk 1]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (rachitnigam) [inline ntt/ntt-cooley-tukey.expect:1]: If naive and `cooley-tukey` generate the same outputs, then there should be one expect file. Use the `expect_dir` option in `runt.toml` to specify that both tests use the same expect file. You'll have to rename the Dahlia file to have the same name (put the two implementation in new folders).

Comment (cgyurgyik) [reply ntt/ntt-cooley-tukey.expect:1]: Just to confirm, you're saying have two folders `cooley-tukey/ntt.fuse` and `naive/ntt.fuse` and then place `ntt.data` in the path designated by `expect_dir`?

**Comment by rachitnigam** [comment]
Looks good. Rename the runt test suite to “NTT correctness” (“correctness” tests use the simulator.

**Comment by cgyurgyik** [comment]
Done

**Pull request merged**